### ISSUE TITLE
Dynamic User Credentials

### ISSUE THREAD

**Issue by icsdevs** [jlevers/selling-partner-api#1]
When using dynamic user credentials, the library should not still try to load from the .env file.

**Comment by jlevers** [comment]
Thanks for opening an issue around this. It's an interesting problem -- the problem is that _some_ of the environment vars are still needed when using dynamic credentials, depending on which credentials the user passes to the `Configuration` object. I'll work on making the `env.php` load environment vars more intelligently (i.e., only loading the vars that are needed for that particular call, and not the ones that aren't).

**Comment by jlevers** [comment]
Interesting, I hadn't thought of that. I'll think about restructuring that part of the library to make it more BYOE (Bring Your Own Environment) friendly.

I really appreciate the feedback!

**Comment by BrockleyJohn** [comment]
I have a related requirement. Building an amazon connector for ecommerce stores and want to use the admin interface for settings instead of editing files. I have managed to work around it by creating a dummy .env file and dynamically setting the $_ENVs in the plugin but it's kinda ugly.

**Comment by jlevers** [comment]
Good to know, thanks. This issue is high priority for the next major release of the library.

**Comment by Nilanth** [comment]
@jlevers How to dynamically set SP API end point and region ?

**Comment by BrockleyJohn** [comment]
with an ugly workaround, e.g.
```
    $endpt = sewc_amazon::get_endpoint(EZ_AMAZON_SPAPI_AWS_REGION);
    $_ENV['SPAPI_ENDPOINT'] = $endpt;
    $this->cfg = new Configuration($authinfo, $endpt);
    ...
    $this->reportAPI = new ReportsApi($this->cfg);
```

**Comment by jlevers** [comment]
Fixed in [`2.1.0`](https://github.com/jlevers/selling-partner-api/releases/tag/v2.1.0).
